### PR TITLE
gracepotma-carbon-doc-suggestions

### PR DESCRIPTION
**Pull request by gracepotma**
Draft content for the carbon section of the Dev Guide. Totally open to changes and I know this doesn't include the more technical dev guidance we still need - just thought it would be good to have some starter content (and this is something I could actually help with a little).

### CODE REVIEW

File: docs/main/carbon [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Using Carbon and the Styleguide
```
Comment (FlorianRappl) [inline docs/main/carbon:1]: Looks good to me - thanks!

Can we name the file "carbon.md"?

**Review by brandones — APPROVED** [review]
Awesome, thanks so much Grace! Eventually I suspect that some of this should be part of the project-plan-type docs on the Wiki, and referenced from here? But glad to have this written anyway.

**Comment by gracepotma** [comment]
> Awesome, thanks so much Grace! Eventually I suspect that some of this should be part of the project-plan-type docs on the Wiki, and referenced from here? But glad to have this written anyway.

Thanks @brandones! Yeah, FWIW the logic/background of Carbon, Design Systems vs Style Guides, etc etc is all documented in the Wiki; thought we may want some brief background in this guide since changing the UI is rather controversial. However definitely feel free to cut out any text you don't find adds value (for devs). I'll rearrange it so that the dev-actionable pieces are at the start rather than the end.

**Comment by brandones** [comment]
Sounds good. What do you mean by "changing the UI is rather controversial," though? Not disagreeing, just clarifying.

**Pull request merged**